### PR TITLE
ref: Migrate isFreePlan to GQL Field and remove Helper

### PR DESCRIPTION
**Pull request by ajay-sentry**
# Description

This PR aims to remove the isFreePlan helper function in Gazebo in favor of the new GraphQL field for the same.

Similar PR to https://github.com/codecov/gazebo/pull/3576. 

One piece of milestone 2: https://github.com/codecov/engineering-team/issues/2989

Closes https://github.com/codecov/engineering-team/issues/2997

Clearly I lied when I said the isEnterprisePlan PR was going to be the largest, each one looks to be similarly difficult though now with 2 under our belts I think we know what each of these will entail for the rest.

**Notes**

- A couple bug fixes included that I found while doing this migration, I'll add comments for them on the review.
- Most of the PR is just swapping over to the new GQL field and updating tests to reflect that new field
- In the cases where useAccountDetails is no longer needed, we did a "hot swap" of usePlanData where applicable. I'll note those areas on the PR as well.

# Screenshots

# Link to Sample Entry

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/pages/MembersPage/MembersList/MembersList.jsx [hunk 1]
```diff
@@ -40,11 +39,11 @@ function useActivateUser({ provider, owner }) {
   return { activate, ...rest }
 }
 
-const handleActivate = (accountDetails, activate, setIsOpen) => (user) => {
+const handleActivate = (planData, activate, setIsOpen) => (user) => {
   if (
-    accountDetails?.activatedUserCount >= accountDetails?.plan?.quantity &&
+    !planData?.plan?.hasSeatsLeft &&
```
Comment (ajay-sentry) [inline src/pages/MembersPage/MembersList/MembersList.jsx:44]: We're using hasSeatsLeft in favor of accountDetails business logic here -- cleaning up the code a bit imo :)

File: src/pages/MembersPage/MembersList/MembersList.jsx [hunk 2]
```diff
@@ -63,7 +62,7 @@ function MembersList() {
   })
 
   const { activate } = useActivateUser({ owner, provider })
-  const { data: accountDetails } = useAccountDetails({ owner, provider })
+  const { data: planData } = usePlanData({ owner, provider })
```
Comment (ajay-sentry) [inline src/pages/MembersPage/MembersList/MembersList.jsx:65]: hot swap of useAccountDetails -> usePlanData

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/FreePlanCard/FreePlanCard.test.jsx [hunk 3]
```diff
@@ -285,20 +276,20 @@ describe('FreePlanCard', () => {
...
     it('renders actions billing button', async () => {
-      setup()
+      setup({})
 
       render(<FreePlanCard plan={freePlan} />, {
         wrapper,
       })
 
-      const link = await screen.findByRole('link', { name: /Manage plan/ })
+      const link = await screen.findByRole('link', { name: /Upgrade/ })
```
Comment (ajay-sentry) [inline src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/FreePlanCard/FreePlanCard.test.jsx:285]: This is actually a bug fix test update

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx [hunk 4]
```diff
@@ -281,7 +284,8 @@ export const useInfiniteTestResults = ({
             data?.owner?.repository?.testAnalytics?.testResults?.totalCount ??
             0,
           private: data?.owner?.repository?.private ?? null,
-          plan: data?.owner?.plan?.value ?? null,
+          planName: data?.owner?.plan?.value ?? null,
```
Comment (ajay-sentry) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx:287]: renaming this field to be more relevant to what it actually is

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/shared/ActionsBilling/ActionsBilling.jsx [hunk 5]
```diff
@@ -62,7 +60,7 @@ function PlansActionsBilling({ plan }) {
     return (
       <div className="flex self-start">
         <Button to={{ pageName: 'upgradeOrgPlan' }} variant="primary">
-          {isSentryPlan(plan?.value) ? 'Manage plan' : 'Upgrade'}
```
Comment (ajay-sentry) [inline src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/shared/ActionsBilling/ActionsBilling.jsx:65]: this wasn't the plan object but rather the plan name so this was never working

File: src/shared/utils/billing.ts [hunk 6]
```diff
@@ -35,13 +35,6 @@ export interface Plan {
   quantity?: number
 }
 
-export function isFreePlan(plan?: PlanName | null) {
```
Comment (RulaKhaled) [inline src/shared/utils/billing.ts:38]: beauty

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx [hunk 7]
```diff
@@ -65,6 +65,7 @@ const GetTestResultsSchema = z.object({
       plan: z
         .object({
           value: z.nativeEnum(Plans),
+          isFreePlan: z.boolean(),
```
Comment (RulaKhaled) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx:68]: is this nullable?

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx [hunk 8]
```diff
@@ -181,7 +183,8 @@ interface UseTestResultsArgs {
     testResults: TestResult[]
     pageInfo: { endCursor: string | null; hasNextPage: boolean }
     private: boolean | null
-    plan: PlanName | null
+    planName: PlanName | null
+    isFreePlan: boolean | null
```
Comment (RulaKhaled) [inline src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx:187]: because it says | null here

**Review by RulaKhaled — APPROVED** [review]
lgtm in general just one question regarding nullable values

File: src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx (hunk 8 above)
Comment (ajay-sentry) [reply src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useInfiniteTestResults/useInfiniteTestResults.tsx:187]: guh good catch, I'll update. It shouldn't be nullable iirc

**Pull request merged**